### ISSUE TITLE
[Question] installed from git. When run writes to terminal: not a git repository before starting

### ISSUE THREAD

**Issue by Nunuvin** [ranger/ranger#1577]
#### Runtime Environment

- Operating system and version:  CentOS6
- Terminal emulator and version: gnome 2.
- Python version:  2.6
- Ranger version/commit: current
- Locale: 

#### Current Behavior
when run in bash as ranger
writes: not a git repository if not in a git repo, then runs
#### Expected Behavior
just to run and open up without attempting to run git
#### Possible Solutions
Not sure where to look. Does ranger invoke git command on startup?

#### Steps to reproduce
1.  git clone
2.  python setup.py install line
3.  ranger

**Comment by toonn** [comment]
Only time ranger calls git is when running `ranger --version` or when git integration's enabled. The version thing's to provide us with information on what particular commit someone's running if they're running ranger from the repo (which is common among those who report bugs).

Do you have git integration enabled? I can't reproduce this.

**Comment by WoolridgeGrant** [comment]
Hello, same issue here : 

- Operating system and version: archlinux
- Terminal emulator : termite
- Python version: 3.7.3
- Ranger version/commit: 1.9.2.4 (from the [AUR)](https://aur.archlinux.org/packages/ranger-git/)

No git integration enabled.
When running ranger in an actual git repository I get the following output : 
```
fatal: No names found, cannot describe anything.
```

**Comment by Alexey104** [comment]
I'm having absolutely the same issue, and i found that i don't get any error messages if i run ranger inside ranger's master repo cloned to my hdd.
But when running ranger outside of this repo i get: "fatal: not a git repository (or any of the parent directories): .git" message, or:


> When running ranger in an actual git repository I get the following output :
> 
> ```
> fatal: No names found, cannot describe anything.
> ```

**Comment by toonn** [comment]
I think what's happening is the `git describe` we use is run in ranger's cwd rather than the directory the code is in.

**Comment by Nunuvin** [comment]
Yeah I think I was using python 2 for ranger (its a 32bit pc and not that many linux distros still support it).
Glad to hear that a new commit should fix it. OS is centos 6.

**Comment by toonn** [comment]
It'd be great if someone could test the fix in #1583.

**Comment by Alexey104** [comment]
> It'd be great if someone could test the fix in #1583.

Thanks for that, but it didn't solve the issue for me. I'm using Arch, python version is 3.7.3.

**Comment by toonn** [comment]
I still can't reproduce this. Could someone describe exactly what they're doing and the errors output?

I *do* get erroneous output when starting ranger from a different git repo, but that should be fixed by #1583.

**Comment by toonn** [comment]
Just a thought: none of the parent directories of the directories this happens in are git repos, are they?

**Comment by Alexey104** [comment]
> Could someone describe exactly what they're doing and the errors output?

Sorry, my English is far from perfect, but i'll try to explain:
When running ranger from the terminal, i see this message:
![Screenshot from 2019-06-05 17-29-30](https://user-images.githubusercontent.com/38672094/58965116-b6ac7b00-87b8-11e9-9e3c-7afb7358925c.png)

When running ranger inside any actual git repo except for ranger's master repo, i see this:
![Screenshot from 2019-06-05 17-40-51](https://user-images.githubusercontent.com/38672094/58965333-1d319900-87b9-11e9-87d5-b2c94066f9ca.png)

After that ranger starts up and works just fine. If running inside ranger's master repo, no any error messages appear.
I have applied #1583 and reinstall ranger manually with this patch, but the issue still exists. It's not a big issue, because ranger works fine after startup, but i haven't gotten these error messages when using ranger before, its appeared recently after some updates.

**Comment by u91o** [comment]
I'm running ranger.py directly without installing and that works fine and dandy. I successfully reproduced the bug by installing from setup.py, however.

**Comment by Alexey104** [comment]
> Could someone verify this?

Yes, confirm.

**Comment by lacygoill** [comment]
FWIW, I think I have the same issue.

Usually, I update ranger by pulling the commits with `$ git pull` then run `$ python3 -m pip install --user --upgrade .`:

    $ python3 -m pip install --user --upgrade .
    Processing /home/user/GitRepos/ranger
    Building wheels for collected packages: ranger-fm
      Building wheel for ranger-fm (setup.py) ... done
      Stored in directory: /tmp/pip-ephem-wheel-cache-0xzk9vp_/wheels/20/cb/0f/083c79d922865cc9e5e42c8f876b8e65c3b9ac6fa4193960ad
    Successfully built ranger-fm
    Installing collected packages: ranger-fm
      Found existing installation: ranger-fm 1.9.2
        Uninstalling ranger-fm-1.9.2:
          Successfully uninstalled ranger-fm-1.9.2
    Successfully installed ranger-fm-1.9.2

It works, but since the commit [`2dad0b8`](https://github.com/ranger/ranger/commit/2dad0b8b4ee6832aff39949e82ed419244abf397), whenever I start ranger, the latter writes this message on its stderr:

    fatal: not a git repository (or any of the parent directories): .git

I tried to replace [this line](https://github.com/ranger/ranger/blob/master/ranger/__init__.py#L27), with:

    stderr=subprocess.PIPE)

Then I ran `$ python3 -m pip install --user --upgrade .`, but the installation fails:

    $ python3 -m pip install --user --upgrade .
    Processing /home/user/GitRepos/ranger
        ERROR: Complete output from command python setup.py egg_info:
        ERROR: v1.9.2-235-g4486ecb9
        Traceback (most recent call last):
          File "<string>", line 1, in <module>
          File "/tmp/pip-req-build-6c9e_555/setup.py", line 12, in <module>
            import ranger
          File "/tmp/pip-req-build-6c9e_555/ranger/__init__.py", line 51, in <module>
            VERSION = version_helper()
          File "/tmp/pip-req-build-6c9e_555/ranger/__init__.py", line 29, in version_helper
            version_string = version_string.format(git_description.strip('\n'))
        AttributeError: 'NoneType' object has no attribute 'strip'
        ----------------------------------------
    ERROR: Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-req-build-6c9e_555/

---

Here's my environment:

   - Operating system and version: `Linux ubuntu 4.15.0-51-generic #55~16.04.1-Ubuntu SMP Thu May 16 09:24:37 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux`
   - Terminal emulator and version: st 0.8.2
   - Python version: 3.7.3
   - Ranger version/commit: 4486ecb
   - Locale: en_US.UTF-8

**Comment by zzjjzzgggg** [comment]
It is strange that version string is obtained by running `git describe` in [__init__.py](https://github.com/ranger/ranger/blob/master/ranger/__init__.py). If I installed ranger by running `make install`, the installed library is not a git repository.

**Comment by toonn** [comment]
@zzjjzzgggg, it's just an opportunistic way of giving *us* the maintainers a *much* clearer indication of what version someone's running. It does no harm, if it fails it just falls back to the hardcoded version number. The only problem is its error messages are currently escaping into user-facing output.

**Comment by toonn** [comment]
@lacygoill, could you try adding that `stderr` line, rather than outright replacing `stdout`?

**Comment by lacygoill** [comment]
I can confirm that adding this line fixes the issue:

    stderr=subprocess.PIPE,

Thank you very much @toonn.

**Comment by toonn** [comment]
It'd be great if people could test the new PR #1584.

**Comment by Alexey104** [comment]
@toonn, it fixed the issue for me.
Many thanks!
